iroh-net/src/endpoint.rs

```diff
@@ -87,117 +104,84 @@ impl Default for Builder {
 }
 
 impl Builder {
-    /// Set a secret key to authenticate with other peers.
+    /// Sets a secret key to authenticate with other peers.
     ///
-    /// This secret key's public key will be the [PublicKey] of this endpoint.
+    /// This secret key's public key will be the [`PublicKey`] of this endpoint and thus
+    /// also its [`NodeId`][]
     ///
     /// If not set, a new secret key will be generated.
     pub fn secret_key(mut self, secret_key: SecretKey) -> Self {
         self.secret_key = Some(secret_key);
         self
     }
 
-    /// Set the ALPN protocols that this endpoint will accept on incoming connections.
-    pub fn alpns(mut self, alpn_protocols: Vec<Vec<u8>>) -> Self {
-        self.alpn_protocols = alpn_protocols;
-        self
-    }
-
-    /// Set an explicit proxy url to proxy all HTTP(S) traffic through.
-    pub fn proxy_url(mut self, url: Url) -> Self {
-        self.proxy_url.replace(url);
-        self
-    }
-
-    /// Set the proxy url from the environment, in this order:
+    /// Set the [ALPN] protocols that this endpoint will accept on incoming connections.
     ///
-    /// - `HTTP_PROXY`
-    /// - `http_proxy`
-    /// - `HTTPS_PROXY`
-    /// - `https_proxy`
-    pub fn proxy_from_env(mut self) -> Self {
-        self.proxy_url = proxy_url_from_env();
-        self
-    }
-
-    /// If *keylog* is `true` and the KEYLOGFILE environment variable is present it will be
-    /// considered a filename to which the TLS pre-master keys are logged.  This can be useful
-    /// to be able to decrypt captured traffic for debugging purposes.
-    pub fn keylog(mut self, keylog: bool) -> Self {
-        self.keylog = keylog;
-        self
-    }
-
-    /// Skip verification of SSL certificates from relay servers
+    /// Not setting this will still allow creating connections, but to accept incoming
+    /// connections the [ALPN] must be set.
     ///
-    /// May only be used in tests.
-    #[cfg(any(test, feature = "test-utils"))]
-    pub fn insecure_skip_relay_cert_verify(mut self, skip_verify: bool) -> Self {
-        self.insecure_skip_relay_cert_verify = skip_verify;
+    /// [ALPN]: https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation
+    pub fn alpns(mut self, alpn_protocols: Vec<Vec<u8>>) -> Self {
+        self.alpn_protocols = alpn_protocols;
         self
     }
 
     /// Sets the relay servers to assist in establishing connectivity.
     ///
-    /// relay servers are used to discover other peers by [`PublicKey`] and also help
-    /// establish connections between peers by being an initial relay for traffic while
-    /// assisting in holepunching to establish a direct connection between peers.
+    /// Relay servers are used to establish initial connection with an other iroh-net node.
```

Comment:
```suggestion
    /// Relay servers are used to establish initial connection with another iroh-net node.
```
this is probably correct but to me it sounds more.. normal?